tasty-dejafu/Test/Tasty/DejaFu.hs

```diff
@@ -246,6 +247,24 @@ testDejafusWay :: Show b
   -> TestTree
 testDejafusWay = testconc (const Nothing)
 
+-- | Variant of 'testDejafusWay' which can selectively discard results.
+--
+-- @since unreleased
+testDejafusDiscard :: Show b
+  => (Either Failure a -> Maybe Discard)
+  -- ^ Selectively discard results.
+  -> Way
+  -- ^ How to execute the concurrent program.
+  -> MemType
+  -- ^ The memory model to use for non-synchronised @CRef@ operations.
+  -> [(TestName, ProPredicate a b)]
+  -- ^ The list of predicates (with names) to check.
+  -> Conc.ConcIO a
+  -- ^ The computation to test.
+  -> TestTree
+testDejafusDiscard discard way memtype tests =
+  testconc discard way memtype tests
```

Comment:
Could you eta-reduce this to make HLint happy?

Reply:
Of course, fixed.